cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -404,6 +417,8 @@ predicate simpleLocalFlowStep(Node nodeFrom, Node nodeTo) {
   simpleInstructionLocalFlowStep(nodeFrom.asInstruction(), nodeTo.asInstruction())
 }
 
+private predicate hasSize(Type t, int size) { t.getSize() = size }
```

Comment:
Why does this need to exist? Doesn't it just get inlined?

Reply:
You're right. It does get inlined. I was trying to prevent a cartesian product, but this predicate just gets inlined and still produce a bad join order (which is responsible for the slowdown in the projects I noted in the CPP-differences). I'm testing a different approach now.

Reply:
Fixed in e0cd595d54abcc5e6c122e85e9d2bdb9d1e95cc8.

Reply:
I've pushed a join order fix now. Once we get the `must-exactly-overlap-with-different-type` overlap we can avoid the extra case in `simpleInstructionLocalFlowStep` altogether.